doc/stdenv/platform-notes.chapter.md

```diff
@@ -2,242 +2,317 @@
 
-  If your derivation has multiple versions of this hook in its inputs (e.g., because it has been propagated by one of your dependencies), it will use the highest deployment target available.
+The current default versions of the deployment target (minimum version) and SDK are indicated by Darwin-specific attributes on the platform. Because of the ways that minimum version and SDK can be changed that are not visible to Nix, they should be treated as lower bounds.
+If you need to parameterize over a specific version, create a function that takes the version as a parameter instead of relying on these attributes.
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3"; # Upstream specifies the minimum supported version as 10.15.
-    buildInputs = [ qt6.qtbase (darwinMinVersionHook "10.15") ];
-  }
-  # Qt 6 specifies a minimum version of 12.0, so the minimum version resolves to 12.0.
-  ```
+- `darwinMinVersion` defaults to 10.12 on x86_64-darwin and 11.0 on aarch64-darwin.
+  It sets the default deployment target.
+- `darwinSdkVersion` defaults to 10.12 on x86-64-darwin and 11.0 on aarch64-darwin.
+  Only the major version determines the SDK version, resulting in the 10.12.2 and 11.3 SDKs being used on these platforms respectively.
 
+#### How to use libiconv on Darwin {#sec-darwin-troubleshooting-libiconv}
 
-- You should rely on the default SDK when possible. If a package specifies a required SDK version, use that version (e.g., libuv requires 11.0, so it should use `apple-sdk_11`). When a package supports multiple SDKs, determine which SDK package to use based on the following rules of thumb:
+The libiconv package is included in the SDK by default along with libresolv and libsbuf.
+You do not need to do anything to use these packages. They are available automatically.
+If your derivation needs the `iconv` binary, add the `libiconv` package to your `nativeBuildInputs` (or `nativeCheckInputs` for tests).
 
-  - If a package supports multiple SDK versions, use the lowest supported SDK version by the package (but no lower than the default SDK). That ensures maximal platform compatibility for the package.
+#### Accessing libutil headers {#sec-darwin-troubleshooting-libutil}
 
-  - If a package specifies a range of supported SDK versions _and_ a minimum supported version, assume the package is using availability checks to support the indicated minimum version. Add the highest supported SDK and a `darwinMinVersionHook` set to the minimum version supported by the upstream package.
+Some packages only need to link `libutil.dylib`.
+A link to the dylib for it is included in the SDK.
+If a package needs the libutil headers, include `darwin.libutil` in your `buildInputs`.
+The `darwin.libutil` package is not propagated by default because the headers are not included by default in the upstream SDK.
+Adding them to the SDK confuses some packages and breaks their builds (e.g., Python).
 
```

Comment:
There’s not a single package in the tree that wants `<libutil.h>` on Darwin other than Apple source releases, and in fact it seems like giving build systems `<libutil.h>` sometimes confuses them. I think we can elide this as more likely to lead people down blind alleys than to help, now that the PR to propagate the `darwin.libutil` library has been merged. (True, there’ll be a lag of a few weeks, but it’s not worth revising the documentation again.)